apps/src/sites/studio/pages/levels/_content.js

```diff
@@ -13,13 +14,28 @@ $(document).ready(() => {
   Blockly.Css.inject(document);
   // Install the common Blockly blocks
   commonBlocks.install(window.Blockly, {});
-  try {
-    // TODO [FND-972] Only the associated level type's blocks.
-    // Install the custom CDO blocks for the associated level type.
-    const appBlocks = require('@cdo/apps/turtle/blocks');
-    appBlocks.install(window.Blockly, {skin: {}, app: 'turtle'});
-  } catch (error) {
-    console.warn(`Unable to load blockly blocks for turtle: ${error}`);
+  if (document.querySelector('script[data-associatedBlocks]')) {
+    const associatedBlocks = getScriptData('associatedblocks');
+    try {
+      // Install the custom CDO blocks for the associated level type.
+      const appBlocks = require(`@cdo/apps/${associatedBlocks}/blocks`);
+      let skin = {};
+      // Some apps require an app skin to be defined. Currently all of these
+      // can use a skin that has the same name as the app itself. In the future,
+      // we might need to allow specifying which skin to use or define a default.
+      if (['bounce', 'flappy', 'jigsaw', 'studio'].includes(associatedBlocks)) {
+        const appSkins = require(`@cdo/apps/${associatedBlocks}/skins`);
+        skin = appSkins.load(assetUrl, associatedBlocks);
+      }
+      appBlocks.install(window.Blockly, {
+        skin,
+        app: associatedBlocks
+      });
+    } catch (error) {
+      console.error(
```

Comment:
I'd love to also log this error to someplace we can monitor it. NewRelic, perhaps?